### ISSUE TITLE
Dead module with an enum causes internal compiler error

### ISSUE THREAD

**Issue by e-kayrakli** [chapel-lang/chapel#6536]
### Summary of Problem
As in the title: dead module with an enum causes internal compiler error

### Steps to Reproduce

**Source Code:**

deadModule.chpl:
```chapel
enum Foo {BAR};
```
mainModule.chpl:
```chapel
proc main() { writeln("Hello"); }
```

**Compile command:**
```
chpl deadModule.chpl mainModule.chpl
```

**Execution command:**
Doesn't compile.

**Associated Future Test(s):**
PR: #6537 

Will add the future `test/users/engin/deadModuleWithEnum/deadModuleWithEnum.chpl`

### Configuration Information

- Output of `chpl --version`:
```
chpl Version 1.16.0 pre-release (ad166c86f8)
Copyright (c) 2004-2017, Cray Inc.  (See LICENSE file for more details)
```

- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
```
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_MAKE: make
CHPL_ATOMICS: intrinsics
CHPL_GMP: gmp
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_WIDE_POINTERS: struct
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:
```
gcc (Ubuntu 6.3.0-12ubuntu2) 6.3.0 20170406
```

**Comment by e-kayrakli** [comment]
For background: @LouisJenkinsCS hit this issue with `BigInteger`. Later he figured out that some enum in the module was causing the error.

**Comment by bradcray** [comment]
This is hitting some logic in deadCodeElimination.cpp that seems to be saying "if there was only one thing in this module, I expected it to be an init function!"  

I'm wondering if @ronawho or @noakesmichael could take a look at this -- Elliot was the original author, and Mike did an edit in the vicinity recently related to determining the main module (I didn't go back to see if the segfault existed prior to that change).  @lydia-duncan has also been here to add the internal error that we're hitting and avoid a segfault, so may have additional insight.

**Comment by ronawho** [comment]
@noakesmichael mind taking an initial look? (I'm out next week, and so it'll be a while until I get a chance to dig into this)

**Comment by lydia-duncan** [comment]
I think @noakesmichael is the right person to look at this.  I suspect his changes to scopeResolve's module handling caused the bug that was fixed last year to resurface

**Comment by lydia-duncan** [comment]
Err, extended the bug that it appears has not been fixed to also apply to enums

**Comment by lydia-duncan** [comment]
Err - either extended the bug, or copied the source of the bug over and it is only just now becoming apparent that it also applies to enums

**Comment by noakesmichael** [comment]
Fixed by #6570
